lib/active_model_serializers/logging.rb

```diff
@@ -0,0 +1,112 @@
+##
+# ActiveModelSerializers::Logging
+#
+#   https://github.com/rails/rails/blob/280654ef88/activejob/lib/active_job/logging.rb
+#
+module ActiveModelSerializers::Logging
+  extend ActiveSupport::Concern
+
+  included do
+    include ActiveModelSerializers::Callbacks
+    extend NotificationMacro
+    instrument_rendering
+  end
+
+  module ClassMethods
+    def instrument_rendering
+      around_render do |args, block|
+        tag_logger do
+          notify_render do
+            block.call(args)
+          end
+        end
+      end
+    end
+  end
+
+  # Adapted from:
+  #   https://github.com/rubygems/rubygems/blob/cb28f5e991/lib/rubygems/deprecate.rb
+  # Provides a single method +notify+ to be used to declare when
+  # something a method notifies, with the argument +callback_name+ of the notification callback.
+  #
+  #     class Adapter
+  #       def self.klass_method
+  #         # ...
+  #       end
+  #
+  #       def instance_method
+  #         # ...
+  #       end
+  #
+  #       include ActiveModelSerializers::Logging
+  #       notify :instance_method, :render
+  #
+  #       class << self
+  #         extend ActiveModelSerializers::Logging::NotificationMacro
+  #         notify :klass_method, :render
+  #       end
+  #     end
+  module NotificationMacro
```

Comment:
Yeah, but since it's really just a way to make sure as_json -> serializable_hash -> to_json doesn't notify three times... and I'm not sure it deserves it's own file right now...